tests-e2e/core/gitopsDeploymentConditions_test.go

```diff
@@ -0,0 +1,223 @@
+package core
```

Comment:
I think file name can be changed to `gitOpsDeployment_test.go` to align with functionality it is testing.
`gitopsDeploymentConditions_test.go` would be conflicting with file for `gitOpsDeployment.status.conditions` tests.